pandas/core/arrays/period.py

```diff
@@ -539,14 +769,154 @@ def _check_timedeltalike_freq_compat(self, other):
                                     .format(cls=type(self).__name__,
                                             freqstr=self.freqstr))
 
+    def repeat(self, repeats, *args, **kwargs):
+        """
+        Repeat elements of a Categorical.
+
+        See also
+        --------
+        numpy.ndarray.repeat
+        """
+        # TODO: Share with Categorical.repeat?
+        # need to use ndarray_values in Categorical
+        # and some kind of _constructor (from_ordinals, from_codes).
+        nv.validate_repeat(args, kwargs)
+        values = self._ndarray_values.repeat(repeats)
+        return type(self)(values, self.freq)
+
+    # Delegation...
+    def strftime(self, date_format):
+        return self._format_native_types(date_format=date_format)
+
+    def astype(self, dtype, copy=True):
+        # TODO: Figure out something better here...
+        # We have DatetimeLikeArrayMixin ->
+        #     super(...), which ends up being... DatetimeIndexOpsMixin?
+        # this is complicated.
+        # need a pandas_astype(arr, dtype).
+        from pandas import Categorical
+
+        dtype = pandas_dtype(dtype)
+
+        if is_object_dtype(dtype):
+            return np.asarray(self, dtype=object)
+        elif is_string_dtype(dtype) and not is_categorical_dtype(dtype):
+            return self._format_native_types()
+        elif is_integer_dtype(dtype):
+            return self.values.astype("i8", copy=copy)
+        elif (is_datetime_or_timedelta_dtype(dtype) and
+              not is_dtype_equal(self.dtype, dtype)) or is_float_dtype(dtype):
+            # disallow conversion between datetime/timedelta,
+            # and conversions for any datetimelike to float
+            msg = 'Cannot cast {name} to dtype {dtype}'
+            raise TypeError(msg.format(name=type(self).__name__, dtype=dtype))
+        elif is_categorical_dtype(dtype):
+            return Categorical(self, dtype=dtype)
```

Comment:
you can do the import here

Reply:
Why's that? pandas will be cached at this point, so it's quite fast.

Reply:
just easier to read